src/ess/reduce/time_of_flight/eto_to_tof.py

```diff
@@ -520,14 +535,57 @@ def _time_of_flight_data_events(
     return da.bins.assign_coords(tof=_bins_no_validate(**parts))
 
 
-def time_of_flight_data(
-    da: RawData,
+def extract_detector_ltotal(
+    detector_beamline: CalibratedBeamline[RunType],
+) -> DetectorLtotal[RunType]:
+    """
+    Extract Ltotal from the detector data.
+    """
+    graph = scn.conversion.graph.beamline.beamline(scatter=True)
+    return DetectorLtotal[RunType](
```

Comment:
I suppose what this really does is the naive straight-line approximation to Ltotal based on basic component positions. I guess the idea is to swap out this provider if something more sophisticated is needed? Maybe the nature of the approximation should be reflected in the names and docstrings?